library/core/src/slice/index.rs

```diff
@@ -231,11 +257,18 @@ unsafe impl<T> SliceIndex<[T]> for usize {
         // `self` is in bounds of `slice` so `self` cannot overflow an `isize`,
         // so the call to `add` is safe.
         unsafe {
-            assert_unsafe_precondition!(
-                "slice::get_unchecked requires that the index is within the slice",
-                [T](this: usize, slice: *const [T]) => this < slice.len()
-            );
-            slice.as_ptr().add(self)
+            #[cfg(bootstrap)]
+            {
+                assert_unsafe_precondition!(
+                    "slice::get_unchecked requires that the index is within the slice",
+                    [T](this: usize, slice: *const [T]) => this < slice.len()
+                );
+                slice.as_ptr().add(self)
+            }
+            #[cfg(not(bootstrap))]
+            {
+                slice_index(slice, this)
```

Comment:
why no `assert_insafe_precondition!` here? does mir indexing make the same assertion?